ipfshttpclient/client/base.py

```diff
@@ -167,25 +166,18 @@ def close(self) -> None:
 
 class _inner_func_t(ty_ext.Protocol, ty.Generic[T]):
 	def __call__(self, *args: ty.Any, **kwargs: ty.Any) \
-	    -> ty.Union[ty.List[T], http.StreamDecodeIteratorSync[T], None]:
+	    -> ty.Union[ty.List[T], http.StreamDecodeIteratorSync[T]]:
 		...
 
 
 class _returns_multiple_wrapper2_t(ty_ext.Protocol, ty.Generic[T, R]):
 	@ty.overload
-	def __call__(self, *args: ty.Any, stream: bool = ...,
-	             return_result: utils.Literal_False, **kwargs: ty.Any) -> None:
+	def __call__(self, *args: ty.Any, stream: utils.Literal_True, **kwargs: ty.Any) -> ty.List[R]:
```

Comment:
I'd like to propose that `Literal[True]` and `Literal[False]` not be used until the minimum Python version is `3.8`

There's quite a few typing edge cases to deal with already related to the documentation tools having gaps, and supporting multiple Python versions. This adds overhead for other contributors and maintainers.

What benefit is being gained by using `Literal[t/f]` instead of `bool` ?

Reply:
> I'd like to propose that Literal[True] and Literal[False] not be used until the minimum Python version is 3.8

Do remember that this means end of 2023 for this library at the earliest.

> What benefit is being gained by using Literal[t/f] instead of bool?

In cases like this one, it allows mypy to distinguish between the two boolean values and have conflict-free overloads (otherwise the overload signatures would be identical). Also this overloading thing will have to be present in the public API as well to provide proper return types for methods actually supporting `stream` (only a couple fortunately) and the distinction will again be just the boolean value.

Reply:
Knowing the EOL was 2023 was actually a part of what motivated my comment - another view is that it will be a drag factor on the library for 3 years.

Can the method be refactored another way, for example, having two methods (different names)? If the return type is correlated 1:1 with a true/false boolean, that seems to be a reason to have two separate methods/functions.

Reply:
I think it's fine to have this for a dynamic language like Python and it's been like this for several years already. I think it's not worth redesigning the API just because it makes the typings a bit more verbose (and a bit ugly) in some places.